pkg/pipeline/conntrack/aggregator.go

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright (C) 2022 IBM, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package conntrack
+
+import (
+	"fmt"
+	"math"
+
+	"github.com/netobserv/flowlogs-pipeline/pkg/api"
+	"github.com/netobserv/flowlogs-pipeline/pkg/config"
+	"github.com/netobserv/flowlogs-pipeline/pkg/pipeline/utils"
+	log "github.com/sirupsen/logrus"
+)
+
+// aggregator represents a single aggregate field in a connection. The aggregated values are stored in the connection
+// but managed by the aggregator.
+type aggregator interface {
+	// addField adds an aggregate field to the connection
+	addField(conn connection)
+	// update updates the aggregate field in the connection based on the flow log.
+	update(conn connection, flowLog config.GenericMap, d direction)
+}
+
+type aggregateBase struct {
+	inputField  string
+	outputField string
+	splitAB     bool
+	initVal     float64
+}
+
+type aggregateSum struct{ aggregateBase }
+type aggregateCount struct{ aggregateBase }
+type aggregateMin struct{ aggregateBase }
+type aggregateMax struct{ aggregateBase }
+
+// TODO: think of adding a more complex operation such as Average Packet Size which involves 2 input fields: Bytes/Packets
+
+// newAggregator returns a new aggregator depending on the output field operation
+func newAggregator(of api.OutputField) (aggregator, error) {
```

Comment:
@ronensc The code for `newAggregator` and for aggregator, in general, can be reused also for extract aggregate .... no in this PR but maybe in following ??? do you want to open an issue so we will remember to look on that?